zebra/zebra_evpn.c

```diff
@@ -1423,8 +1423,23 @@ void zebra_evpn_rem_macip_add(vni_t vni, const struct ethaddr *macaddr,
 	 * REMOTE - if ES is not local
 	 */
 	if (flags & ZEBRA_MACIP_TYPE_SYNC_PATH) {
-		zebra_evpn_process_sync_macip_add(zevpn, macaddr, ipa_len,
-						  ipaddr, flags, seq, esi);
+		struct zebra_evpn_es *es;
+
+		es = zebra_evpn_es_find(esi);
+		if (es && (es->flags & ZEBRA_EVPNES_READY_FOR_BGP)) {
+			zebra_evpn_process_sync_macip_add(zevpn, macaddr,
+							  ipa_len, ipaddr,
+							  flags, seq, esi);
+		} else {
+			char esi_str[ESI_STR_LEN];
```

Comment:
Shouldn't these two lines be under the DEBUG flag ?

Reply:
Ack, fixed. And thanks for the review Mohan.